### DISCUSSION TITLE
IP with port in ip_address field on db

### DISCUSSION THREAD

**Discussion by jospenaca** [YOURLS/YOURLS#3067] [Q&A]
Hello there, recently I installed YOURLS in Azure with App Service. I successfully insalled. The is that when short link is accessed, the application save IP as X.X.X.X:yyyy. This issue make COUNTRY CODE data not being saved. I had 64 registries, rarely only 2 got full IP withour port (yyyy) and it shows COUNTRY CODE.
Can I do something to resolve this on my own? Is this an issue in the code?

Let me know any sugguetion In can try in the code to see.
Basically, this affect that stats of TRAFFIC LOCATION, not showing nothing.

Thanks.

Jose.

**Comment by dgw** [comment]
I'm not _really_ sure what's being asked here as I keep stumbling over missing words. However, I'll hazard a guess: This instance of YOURLS is behind an Azure Application Gateway? That could explain the port number being included in the saved IP data, since apparently [Azure inserts it into `X-Forwarded-For` headers](https://azure.microsoft.com/en-us/blog/rewrite-http-headers-with-azure-application-gateway/) for some reason. That page also includes some instruction for removing the unwanted port information.

**Reply by jospenaca** [reply]
My apolgize not being clear with my question... the question is how to tweak the code to remove the port from ip address in order to save only the IP. When IP include port number, function that get country code is not working.

Thanks.

Jose.

**Reply by dgw** [reply]
You should try following the instructions in the link I gave above to configure your Application Gateway not to include the port number in its headers. **Or** if I've misinterpreted your description of the environment, please correct me and I'll look for relevant tips.

It's certainly doable to make a little plugin remove the port number for you, but I'd prefer fixing the host-side configuration if possible as there are some edge cases that would be annoying to deal with in such a plugin.

Making `yourls_get_IP()` and/or `yourls_sanitize_ip()` a bit smarter could be a good Hacktoberfest project for someone.
